internal/packer/gaf.go

```diff
@@ -0,0 +1,123 @@
+package packer
+
+import (
+	"archive/zip"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+// overwriteGaf writes a gaf (gokrazy archive format) file
+// by packing build artifacts and
+// storing them into a newly created, uncompressed zip.
+func (p *Pack) overwriteGaf(root *FileInfo) error {
+	dir, err := os.MkdirTemp("", "gokrazy")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(dir)
+
+	tmpMBR, err := os.Create(filepath.Join(dir, "mbr.img"))
+	if err != nil {
+		return err
+	}
+	defer os.Remove(tmpMBR.Name())
+
+	tmpBoot, err := os.Create(filepath.Join(dir, "boot.img"))
+	if err != nil {
+		return err
+	}
+	defer os.Remove(tmpBoot.Name())
+
+	tmpRoot, err := os.Create(filepath.Join(dir, "root.img"))
+	if err != nil {
+		return err
+	}
+	defer os.Remove(tmpRoot.Name())
+
+	tmpSBOM, err := os.Create(filepath.Join(dir, "sbom.json"))
+	if err != nil {
+		return err
+	}
+	defer os.Remove(tmpSBOM.Name())
+
+	if err := p.writeBoot(tmpBoot, tmpMBR.Name()); err != nil {
+		return err
+	}
+
+	if err := writeRoot(tmpRoot, root); err != nil {
+		return err
+	}
+
+	sbomMarshaled, _, err := GenerateSBOM()
+	if err != nil {
+		return err
+	}
+
+	if _, err := tmpSBOM.Write(sbomMarshaled); err != nil {
+		return err
+	}
+
+	if err := writeGafArchive(dir, p.Output.Path); err != nil {
```

Comment:
Can you explicitly Close() the files (tmpSBOM etc.) before calling writeGafArchive? I think the OS should provide a consistent view of the file when it’s opened multiple times, but better to be sure and avoid these sorts of subtle (cross-operating system) questions…

Reply:
Ahh yeah! Good catch. Fixed.